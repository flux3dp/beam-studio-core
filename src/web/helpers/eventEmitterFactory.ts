import EventEmitter from 'eventemitter3';

const eventEmitters = {
  'flux-id': null,
  'top-bar': null,
  'top-bar-hints': null,
  'time-estimation-button': null,
};
export default {
  createEventEmitter: (type?: string): EventEmitter => {
    if (!type) return new EventEmitter();
    if (!eventEmitters[type]) eventEmitters[type] = new EventEmitter();
    return eventEmitters[type];
  },
};
