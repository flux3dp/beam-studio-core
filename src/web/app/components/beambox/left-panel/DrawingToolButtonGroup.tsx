import React, { useEffect, useState } from 'react';

import browser from 'implementations/browser';
import dialogCaller from 'app/actions/dialog-caller';
import eventEmitterFactory from 'helpers/eventEmitterFactory';
import FnWrapper from 'app/actions/beambox/svgeditor-function-wrapper';
import i18n from 'helpers/i18n';
import LeftPanelIcons from 'app/icons/left-panel/LeftPanelIcons';
import LeftPanelButton from 'app/components/beambox/left-panel/LeftPanelButton';
import { getCurrentUser } from 'helpers/api/flux-id';

const LANG = i18n.lang.beambox.left_panel;

const eventEmitter = eventEmitterFactory.createEventEmitter('drawing-tool');

const DrawingToolButtonGroup = ({ className }: { className: string }): JSX.Element => {
  const [activeButton, setActiveButton] = useState('Cursor');
  const isSubscribed = getCurrentUser()?.info?.subscription?.is_valid;
  const renderToolButton = (
    id: string,
    icon: JSX.Element,
    label: string,
    onClick: () => void,
    showBadge?: boolean
  ) => (
    <LeftPanelButton
      id={`left-${id}`}
      active={activeButton === id}
      title={label}
      icon={icon}
      onClick={() => {
        setActiveButton(id);
        onClick();
      }}
      showBadge={showBadge}
    />
  );

  useEffect(() => {
    eventEmitter.on('SET_ACTIVE_BUTTON', setActiveButton);
    return () => {
      eventEmitter.removeListener('SET_ACTIVE_BUTTON');
    };
  }, []);

  return (
    <div className={className}>
      {renderToolButton(
        'Cursor',
        <LeftPanelIcons.Cursor />,
        `${LANG.label.cursor} (V)`,
        FnWrapper.useSelectTool
      )}
      {renderToolButton(
        'Photo',
        <LeftPanelIcons.Photo />,
        `${LANG.label.photo} (I)`,
        FnWrapper.importImage
      )}
      {renderToolButton(
        'MyCloud',
        <LeftPanelIcons.Cloud />,
        LANG.label.my_cloud,
        () => dialogCaller.showMyCloud(FnWrapper.useSelectTool),
        isSubscribed
      )}
      {renderToolButton(
        'Text',
        <LeftPanelIcons.Text />,
        `${LANG.label.text} (T)`,
        FnWrapper.insertText
      )}
      {renderToolButton('Element', <LeftPanelIcons.Element />, `${LANG.label.shapes} (E)`, () =>
        dialogCaller.showShapePanel(FnWrapper.useSelectTool)
      )}
      {renderToolButton(
        'Rectangle',
        <LeftPanelIcons.Rect />,
        `${LANG.label.rect} (M)`,
        FnWrapper.insertRectangle
      )}
      {renderToolButton(
        'Ellipse',
        <LeftPanelIcons.Oval />,
        `${LANG.label.oval} (L)`,
        FnWrapper.insertEllipse
      )}
      {renderToolButton(
        'Polygon',
        <LeftPanelIcons.Polygon />,
        LANG.label.polygon,
        FnWrapper.insertPolygon
      )}
      {renderToolButton(
        'Line',
        <LeftPanelIcons.Line />,
        `${LANG.label.line} (\\)`,
        FnWrapper.insertLine
      )}
      {renderToolButton(
        'Pen',
        <LeftPanelIcons.Draw />,
        `${LANG.label.pen} (P)`,
        FnWrapper.insertPath
      )}
      {renderToolButton('QRCode', <LeftPanelIcons.QRCode />, LANG.label.qr_code, () =>
        dialogCaller.showQRCodeGenerator(FnWrapper.useSelectTool)
      )}
      {renderToolButton('Boxgen', <LeftPanelIcons.Boxgen />, LANG.label.boxgen, () =>
        dialogCaller.showBoxGen(FnWrapper.useSelectTool)
      )}
      {renderToolButton('DM', <LeftPanelIcons.DM />, 'Design Market', () =>
        browser.open(i18n.lang.topbar.menu.link.design_market)
      )}
    </div>
  );
};

export default DrawingToolButtonGroup;
