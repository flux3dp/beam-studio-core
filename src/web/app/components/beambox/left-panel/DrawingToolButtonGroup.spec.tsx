/* eslint-disable import/first */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import eventEmitterFactory from 'helpers/eventEmitterFactory';

const mockUseSelectTool = jest.fn();
const mockImportImage = jest.fn();
const mockInsertText = jest.fn();
const mockInsertRectangle = jest.fn();
const mockInsertEllipse = jest.fn();
const mockInsertLine = jest.fn();
const mockInsertPath = jest.fn();
const mockInsertPolygon = jest.fn();
jest.mock('app/actions/beambox/svgeditor-function-wrapper', () => ({
  useSelectTool: mockUseSelectTool,
  importImage: mockImportImage,
  insertText: mockInsertText,
  insertRectangle: mockInsertRectangle,
  insertEllipse: mockInsertEllipse,
  insertLine: mockInsertLine,
  insertPath: mockInsertPath,
  insertPolygon: mockInsertPolygon,
}));

const mockOpen = jest.fn();
jest.mock('implementations/browser', () => ({
  open: mockOpen,
}));

const showShapePanel = jest.fn();
const showQRCodeGenerator = jest.fn();
const showBoxGen = jest.fn();
const showMyCloud = jest.fn();
jest.mock('app/actions/dialog-caller', () => ({
  showShapePanel: (...args) => showShapePanel(...args),
  showQRCodeGenerator: (...args) => showQRCodeGenerator(...args),
  showBoxGen: (...args) => showBoxGen(...args),
  showMyCloud: (...args) => showMyCloud(...args),
}));

jest.mock('helpers/i18n', () => ({
  lang: {
    beambox: {
      left_panel: {
        label: {
          cursor: 'Select',
          photo: 'Image',
          text: 'Text',
          line: 'Line',
          rect: 'Rectangle',
          oval: 'Oval',
          polygon: 'Polygon',
          pen: 'Pen',
          shapes: 'Elements',
          boxgen: 'Boxgen',
        },
      },
    },
    topbar: {
      menu: {
        link: {
          design_market: 'https://dmkt.io',
        },
      },
    },
  },
}));

const getCurrentUser = jest.fn();
jest.mock('helpers/api/flux-id', () => ({
  getCurrentUser: () => getCurrentUser(),
}));

import DrawingToolButtonGroup from './DrawingToolButtonGroup';

describe('test DrawingToolButtonGroup', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should render correctly', () => {
    const { container } = render(<DrawingToolButtonGroup className="flux" />);
    expect(container).toMatchSnapshot();

    fireEvent.click(container.querySelector('#left-Photo'));
    expect(container).toMatchSnapshot();
    expect(mockImportImage).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Text'));
    expect(container).toMatchSnapshot();
    expect(mockInsertText).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Rectangle'));
    expect(container).toMatchSnapshot();
    expect(mockInsertRectangle).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Ellipse'));
    expect(container).toMatchSnapshot();
    expect(mockInsertEllipse).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Polygon'));
    expect(container).toMatchSnapshot();
    expect(mockInsertPolygon).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Line'));
    expect(container).toMatchSnapshot();
    expect(mockInsertLine).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Element'));
    expect(container).toMatchSnapshot();
    expect(showShapePanel).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Pen'));
    expect(container).toMatchSnapshot();
    expect(mockInsertPath).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-Cursor'));
    expect(container).toMatchSnapshot();
    expect(mockUseSelectTool).toHaveBeenCalledTimes(1);

    fireEvent.click(container.querySelector('#left-QRCode'));
    expect(container).toMatchSnapshot();
    expect(showQRCodeGenerator).toHaveBeenCalledTimes(1);
    expect(showQRCodeGenerator).toHaveBeenLastCalledWith(mockUseSelectTool);

    fireEvent.click(container.querySelector('#left-DM'));
    expect(container).toMatchSnapshot();
    expect(mockOpen).toHaveBeenCalledTimes(1);
    expect(mockOpen).toHaveBeenLastCalledWith('https://dmkt.io');

    fireEvent.click(container.querySelector('#left-Boxgen'));
    expect(container).toMatchSnapshot();
    expect(showBoxGen).toHaveBeenCalledTimes(1);
    expect(showBoxGen).toHaveBeenCalledWith(mockUseSelectTool);

    fireEvent.click(container.querySelector('#left-MyCloud'));
    expect(container).toMatchSnapshot();
    expect(showMyCloud).toHaveBeenCalledTimes(1);
    expect(showMyCloud).toHaveBeenCalledWith(mockUseSelectTool);
  });

  test('should render flux plus icon correctly', () => {
    getCurrentUser.mockReturnValue({ info: { subscription: { is_valid: true } } });
    const { container } = render(<DrawingToolButtonGroup className="flux" />);

    const myCloudButton = container.querySelector('#left-MyCloud');
    expect(myCloudButton).toMatchSnapshot();

    fireEvent.click(myCloudButton);
    expect(myCloudButton).toMatchSnapshot();
    expect(showMyCloud).toHaveBeenCalledTimes(1);
    expect(showMyCloud).toHaveBeenCalledWith(mockUseSelectTool);
  });

  test('event emitter', async () => {
    const eventEmitter = eventEmitterFactory.createEventEmitter('drawing-tool');
    const { container } = render(<DrawingToolButtonGroup className="flux" />);

    expect(container.querySelector('#left-Cursor')).toHaveClass('active');
    eventEmitter.emit('SET_ACTIVE_BUTTON', 'Pen');
    await new Promise((r) => setTimeout(r, 0));
    expect(container.querySelector('#left-Cursor')).not.toHaveClass('active');
    expect(container.querySelector('#left-Pen')).toHaveClass('active');
  });
});
