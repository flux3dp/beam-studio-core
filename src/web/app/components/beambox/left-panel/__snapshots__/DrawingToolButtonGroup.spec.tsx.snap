// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DrawingToolButtonGroup should render correctly 1`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container active"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 2`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 3`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 4`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 5`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 6`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 7`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 8`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 9`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 10`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container active"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 11`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;

exports[`test DrawingToolButtonGroup should render correctly 12`] = `
<div>
  <div
    class="flux"
  >
    <div
      class="container"
      id="left-Cursor"
      title="Select (V)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Photo"
      title="Image (I)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Text"
      title="Text (T)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Element"
      title="Elements (E)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Rectangle"
      title="Rectangle (M)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Ellipse"
      title="Oval (L)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Polygon"
      title="Polygon"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Line"
      title="Line (\\)"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-Pen"
      title="Pen (P)"
    >
      <svgr-url />
    </div>
    <div
      class="container active"
      id="left-Boxgen"
      title="Boxgen"
    >
      <svgr-url />
    </div>
    <div
      class="container"
      id="left-DM"
      title="Design Market"
    >
      <svgr-url />
    </div>
  </div>
</div>
`;
